### DISCUSSION TITLE
Streamlining

### DISCUSSION THREAD

**Discussion by RMBrouwer** [theislab/sfaira#399] [Q&A] [answered]
Hi, 
I have been trying to use sfaira to streamline datasets; but I'd like to keep track of the genes that were originally in the dataset and those that were not. Streamlining currently adds a lot of genes with zero expression. Filtering on those also removes all the genes that were assessed but not detected in the original dataset. Since the gene-names are changing too, it is hard to keep track of these without 'manually' comparing the different naming systems (which is exactly what I would like to avoid using sfaira). Is there a way of doing this that I missed? 

Thanks for your help, 
Rachel

**Comment by davidsebfischer** [comment, marked as answer]
Hi Rachel, thanks for starting this discussion! How do you imagine the output of the streamlining to look like? e.g:
- Do you not want to streamline the features at all (maybe only meta data?), thus maintaining the original gene names? or
- Do you want to map featuers to a genome assembly that can be matched but not add empty genes from that assembly into the streamlined data set?

**Reply by RMBrouwer** [reply]
Thanks for a quick reply! 

I would like to have your second option so that gene names are consistent across datasets, but that no empty genes are added so that I can distinguish between not measured genes and zero expression. Would that be possible?

**Reply by davidsebfischer** [reply]
I opened an issue here for this here https://github.com/theislab/sfaira/issues/400. Note though, unless you filtered genes, these other genes are observed zeros so actually meaningful measurements.

**Reply by RMBrouwer** [reply]
Hi David, 
Now I am confused (just checking in case you will put time into something that I misunderstood): I did not filter genes and there are observed zeros in the original dataset(s). I would like to keep those. The newly added genes after streamlining are those that were not in my original dataset right? So how can they be observed zeros? Maybe I am not using the right terminology above; with zero expression I meant observed zeros, by not measured genes I meant the genes that were not in the original dataset. 
Thanks,
Rachel

**Reply by davidsebfischer** [reply]
Hi Rachel, it really depends on what happened to the data set before, let's say the .bam was alidned to reference genome X, you then received a count matrix with genes that were observed in at least one cell (this is relatively common). Now, the set difference between the genes in your count matrix and the genes in assembly X is actually genes that were zero in all cells, so true zeros as well. If you overlap your count matrix that originated from an alignment to X, to a genes from a reference assembly Y, then you may get a few all-zero genes that are unique to Y that you dont actually want in your streamlined count matrix.

**Reply by RMBrouwer** [reply]
Ah sure, that makes it clear, thanks!
